chore(infra): dogfooding oxfmt

leaysgur opened the pull request:
Fixes #14803 

- Added `oxfmt` and `oxfmtrc.jsonc`
- Remove `dprint` plugin and useless ignore paths
- Apply `dprint` again
- Apply `oxfmt`

Boshen reviewed (approved):
Probably need to squash the two PRs so they can be merged?

leaysgur committed:
chore(infra): dogfooding oxfmt (#14979)

Fixes #14803

- Added `oxfmt` and `oxfmtrc.jsonc`
- Remove `dprint` plugin and useless ignore paths
- Apply `dprint` again
- Apply `oxfmt`
